vertical-pod-autoscaler/pkg/recommender/util/binary_decaying_histogram.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+)
+
+// NewHistogram returns a new Histogram instance using given options.
+func NewBinaryDecayingHistogram(options HistogramOptions, retentionDays int) Histogram {
+	if retentionDays < 3 {
+		panic("at least 3 days retention is required")
+	}
+	return &binaryDecayingHistogram{
+		options:       options,
+		bucketForDay:  make([]uint16, retentionDays),
+		lastDayIndex:  0,
+		retentionDays: retentionDays}
+}
+
+// A histogram that only supports returning the maximum value (Percentile(1.0)) of the samples in the last retentionDays days.
+// This histogram does not support SubtractSample function.
+type binaryDecayingHistogram struct {
+	// Bucketing scheme.
+	options HistogramOptions
+	// Index of a bucket for each day. Buckets are indexed starting with 1. 0 means no sample, 1 means the first bucket, etc.
```

Comment:
Why are buckets indexed from 1 / what does a bucket at index 0 for no sample entail?